maven/lib/dependabot/maven/requirement.rb

```diff
@@ -32,7 +34,11 @@ def self.requirements_array(requirement_string)
 
       def initialize(*requirements)
         requirements = requirements.flatten.flat_map do |req_string|
-          convert_java_constraint_to_ruby_constraint(req_string)
+          if RUBY_VERSION_RESTRICTIONS.any? { |vr| req_string.include?(vr) }
```

Comment:
I attempted to reverse this and match on `MAVEN_RANGE_REGEX = /[\(\[].*,.*[\)\]]/` but we have a bunch of tests for version that don't match this that still get special maven/gradle treatment so opted for checking against known ruby version range restrictions: https://docs.ruby-lang.org/en/2.6.0/Gem/Requirement.html